File: docs/source/examples/CRUD/Transaction.java
Realm realm = Realm.getDefaultInstance();

// Open a thread-safe transaction.
realm.executeTransaction(r -> {
    // ... Make changes
    // Realm automatically cancels the transaction if this
    // code block throws an exception. Otherwise, Realm
    // automatically commits the transaction at the end
    // of the code block.
});

// ...

// Always remember to close the realm when 
// you are done with it.
realm.close();

File: docs/source/examples/CRUD/Transaction.js
// Open the default realm with the relevant schema.
const realm = await Realm.open({ schema: [DogSchema] });

// Open a transaction.
realm.write(() => {
  // ... Make changes ...
  // Realm automatically cancels the transaction if this
  // code block throws an exception. Otherwise, Realm
  // automatically commits the transaction at the end
  // of the code block.
});
